### ISSUE TITLE
DataTemplateDemo duplicating items when collection changes and not removing removed models

### ISSUE THREAD

**Issue by monkog** [helix-toolkit/helix-toolkit#104]
In the DataTemplateDemo in the ItemsVisual3D class the
private void ItemsSourceChanged(DependencyPropertyChangedEventArgs e)
method is wrong. Despite the fact that it's done manually, it doesn't handle the collection changing in the right way.

Every time the collection changes it adds all items from the collection once again, even though, some of them may already be set as the Children of the Viewport.
Also when you delete items from the collection, they don't get deleted from the Children of the Viewport.

A quick workaround is to add Children.Clear(); on top of the method.

**Comment by mwpowellhtx** [comment]
Wholesale Clear and replacement would be mightily inefficient I think, at least extending outward through rendering. Should consider doing Except with some combination of ReferenceEquals (at least). Unless there are other id's involved that could easily be checked. Of course, all this in a Begin/End-Edit boundary.

**Comment by objorke** [comment]
Right, the ItemsSourceChanged is not implemented correctly. This is a very alpha demo. Also, `INotifyCollectionChanged` events should be handled.
I also agree that a better solution than "wholesale clear and replacement" :-) should be found!
I know @JohanLarsson is working on a prototype that solves this problems in a better way.

**Comment by JohanLarsson** [comment]
My prototype turned into a mess. Hope I will find time to rewrite it soon.
